include/boost/algorithm/is_clamped.hpp

```diff
@@ -0,0 +1,72 @@
+/* 
+   Copyright (c) Marshall Clow 2008-2012.
+
+   Distributed under the Boost Software License, Version 1.0. (See accompanying
+   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+  
+*/
+
+/// \file is_clamped.hpp
+/// \brief IsClamped algorithm
+/// \author TODO
+///
+
+#ifndef BOOST_ALGORITHM_IS_CLAMPED_HPP
+#define BOOST_ALGORITHM_IS_CLAMPED_HPP
+
+#include <functional>       //  For std::less
+#include <cassert>
+
+#include <boost/mpl/identity.hpp>      // for identity
```

Comment:
Instead use `boost::type_identity` from `<boost/type_traits/type_identity.hpp>`

Reply:
Hi Glen, I am not familiar with boost machinery. clamp.hpp (which this code is based on) uses MPL.

What is the reason to prefer one over the other? Maybe MPL works in C++98 so that is why it was used?

Reply:
Yes. My understanding is that MP11 is much lighter weight, compile-wise, but requires C++11.

Reply:
> Instead use `boost::type_identity` from `<boost/type_traits/type_identity.hpp>`

I'm pretty sure that when I wrote `boost::clamp` we didn't have `boost::type_identity`; hence the `mp::Identity`

Reply:
@mclow  to see if I got the comments right:
 I should switch to the `boost::type_identity` both here and in clamp.hpp?

Reply:
I have changed (locally) the `mpl::identity` to `type_identity` in clamp.h and is_sorted.h.  Passes all the tests locally.

Reply:
Ok, should I do same here or you prefer to do the modernize(type_identity change) pass in separate pull request?

Also should  I merge is_clamped.hpp into clamp.hpp or you prefer to keep it separate files?

Reply:
I think you should leave it in a separate file; and yes, please remove the dependency on `Boost::MPL`

See https://github.com/boostorg/algorithm/commit/c5f6f52560545e8bd70e390a70c55dd2ff308d0f if you have any questions.

Reply:
I have removed the dependency on mpl from source, but not sure if there is something in build config I should also remove, some equivalent of target_link_libraries in CMake.